### PR TITLE
Create a is_wic_eligible file

### LINKED ISSUE #4744: Create a is_wic_eligible file
We currently add the eligibility in the main 'wic' variable file but it would be easier for integration testing to split it up and use a `defined_for` in the wic variable

### PR DESCRIPTION
**Pull request by hua7450**
Fixes #4744

### CODE REVIEW

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/tests/policy/baseline/gov/usda/wic/wic.yaml [hunk 1]
```diff
@@ -6,59 +6,47 @@
...
   input:
     wic_category_str: PREGNANT
-    meets_wic_income_test: true
-    meets_wic_categorical_eligibility: false
-    is_wic_at_nutritional_risk: true
+    is_wic_eligible: true
   output:
     wic: 37.33 * 12 # Eligible.
 
 - name: Income-eligible and categorically eligible but not at nutritional risk.
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/usda/wic/wic.yaml:13]: update test names

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/tests/policy/baseline/gov/usda/wic/wic.yaml [hunk 2]
```diff
@@ -1,64 +1,40 @@
...
-    meets_wic_income_test: true
-    meets_wic_categorical_eligibility: true
-    is_wic_at_nutritional_risk: true
+    is_wic_eligible: true
   output:
-    wic: 0 # Ineligible.
+    wic: 0 
 
-- name: At nutritional risk and categorically eligible, check uprating.
+- name: Households that are in breastfeeding category are elgible, check uprating.
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/usda/wic/wic.yaml:33]:
```suggestion
- name: Eligible breastfeeding individuals. Verifies uprating.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
@PavelMakarchuk  is the standard to leave off the is_ and just do wic_eligible?

**Comment by PavelMakarchuk** [comment]
> @PavelMakarchuk is the standard to leave off the is_ and just do wic_eligible?

We have used both in the past, for benefit programs we have used `is_x_eligible` in the past but I think the following structure could be good to adapt:
`is_x_eligible` - person level eligibility 
`x_eligible` - TaxUnit/SpmUnit/Household level eligibility

**Review by MaxGhenis — APPROVED** [review]
Could you elaborate on the rationale for that naming convention in https://github.com/PolicyEngine/policyengine-us/issues/4771?

Merging and we can adopt a standard in the future.

**Pull request merged**